README.md

```diff
@@ -1,29 +1,234 @@
+The Catalog is queryable via an API which runs alongside the Profiles Controller.
+
+Profiles can be added to the Catalog by creating a [`ProfileCatalogSource`](#profile-catalog-source).
+
+### Profile Catalog Source
+
+A `ProfileCatalogSource` is a custom resource through which approved Profiles can be managed in the [Catalog](#catalog)
+
+```go
+// ProfileCatalogSourceSpec defines the desired state of ProfileCatalogSource
+type ProfileCatalogSourceSpec struct {
+	// Profiles is the list of profiles exposed by the catalog
+	Profiles []ProfileDescription `json:"profiles,omitempty"`
+}
+
+// ProfileDescription defines details about a given profile.
+type ProfileDescription struct {
+	// Profile name
+	Name string `json:"name,omitempty"`
+	// Profile description
+	Description string `json:"description,omitempty"`
+	// Version
+	// +optional
+	Version string `json:"version,omitempty"`
+	// CatalogSource is the name of the catalog the profile is listed in
+	// +optional
+	CatalogSource string `json:"catalog,omitempty"`
+	// URL is the full URL path to the profile.yaml
+	// +optional
+	URL string `json:"url,omitempty"`
+	// Maintainer is the name of the author(s)
+	// +optional
+	Maintainer string `json:"maintainer,omitempty"`
+	// Prerequisites are a list of dependencies required by the profile
+	// +optional
+	Prerequisites []string `json:"prerequisites,omitempty"`
+}
+```
+
+Profiles can therefore be grouped and namespaced within the Catalog.
+
+### Profile Subscription
+
+A `ProfileSubscription` is the top-level Profile installation object. Once a Subscription is
+applied to the cluster, the requested Profile is parsed and child artifact objects are created.
+
+Artifact objects are processed by Flux components, with the health of Subscription children
+reflected in the `ProfileSubscriptionStatus`.
+
+```go
+// ProfileSubscriptionSpec defines the desired state of a ProfileSubscription
+type ProfileSubscriptionSpec struct {
+	// ProfileURL is a fully qualified URL to a profile repo
+	ProfileURL string `json:"profileURL,omitempty"`
+	// Branch is the git repo branch containing the profile definition (default: main)
+	// +kubebuilder:default:=main
+	// +optional
+	Branch string `json:"branch,omitempty"`
+
+	// Values holds the values for the Helm chart specified in the first artifact
+	// +optional
+	Values *apiextensionsv1.JSON `json:"values,omitempty"`
+
+	// ValuesFrom holds references to resources containing values for the Helm chart specified in the first artifact
+	// +optional
+	ValuesFrom []helmv2.ValuesReference `json:"valuesFrom,omitempty"`
+}
+
+// ProfileSubscriptionStatus defines the observed state of ProfileSubscription
+type ProfileSubscriptionStatus struct {
+	// Conditions holds the conditions for the ProfileSubscription
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+```
+
+Profiles can only be installed if they are listed in a [Catalog Source](#profile-catalog-source).
```

Comment:
is that true?

Reply:
currently, yes i think

but will change

Reply:
where does it get enforced? is it just in pctl? I think I can create a profilesub manually and it would work for any profile

Reply:
oh i mean if done the "official" way by pctl. i will clarify